src/ir/item.rs

```diff
@@ -181,6 +256,39 @@ impl Item {
             })
     }
 
+    /// Returns the template arguments that apply to a struct. This is a concept
+    /// needed because of type declarations inside templates, for example:
+    ///
+    /// ```c++
+    /// template<typename T>
+    /// class Foo {
+    ///     typedef T element_type;
+    ///     typedef int Bar;
+    ///
+    ///     template<typename U>
+    ///     class Baz {
+    ///     };
+    /// };
+    /// ```
+    ///
+    /// In this case, the applicable template arguments for the different types
+    /// would be:
+    ///
+    ///  * `Foo`: [`T`]
+    ///  * `Foo::element_type`: [`T`]
+    ///  * `Foo::Bar`: [`T`]
+    ///  * `Foo::Baz`: [`T`, `U`]
+    ///
+    /// You might notice that we can't generate something like:
+    ///
+    /// ```rust,ignore
+    /// type Foo_Bar<T> = ::std::os::raw::c_int;
```

Comment:
Yeah, what I meant in my above aside would be generating code like this

```rust
type Foo_Bar<T> = (::std::os::raw::c_int, PhantomData<T>);
```

in this case.